sympy/ntheory/primetest.py

```diff
@@ -110,6 +110,34 @@ def mr(n, bases):
     return True
 
 
+
+
+"""Tests whether n is prime or an Euler pseudoprime to base b."""
+def is_euler_pseudoprime(n, b):
+	if not mr(n, [b]):
+        return False
+    
+    
+	r = n-1
+    r = as_int(r)
```

Comment:
- [ ] indentation is off; I will kill the Travis run.
- [ ] Since you use `n` later, make n an int via `n = as_int(n)`; then calculations done with it will still be int.
- [ ] Use `trailing` as I said before.
- [ ] Don't use `(` after `if` in the remaining tests.
- [ ] Move the `if c == 1: return False` to *after* the recalculation of `c`.

Reply:
Sir ,
Now  I have done following changes to code 
1] I have put the docstring inside the function now according to suggestion by smichr
2] indented properly as previous error occurred due to mix of tabs and space which wasn't visible in github sorry for trouble
3] made n=as_int(n)
4]Used trailing from sympy factor to calculate ending zeros of r instead of while loop
5]removed bracket after if following PEP standards.
6]Moved the if c == 1: return False to after the recalculation of c.
Thanks for help and suggestions.